src/components/SideBar/SideBar.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import {
+    MainList,
+    ListItemLarge,
+    ListItemLargeText,
+    ListItemSmallText,
+    ListItemSmall,
+    IconLarge,
+    SocialList,
+    useSideBarStyles,
+    IconWrapper,
+} from './style';
+
+import CustomImg from '../../components/CustomImg';
+import CustomIcon from '../../components/CustomIcon';
+import logo from '../../assets/img/logo.svg';
+import Drawer from '@material-ui/core/Drawer';
+import List from '@material-ui/core/List';
+import Divider from '@material-ui/core/Divider';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import Tooltip from '@material-ui/core/Tooltip';
+
+import GithubIcon from '../../assets/img/sm_github_icon.svg';
+import RiotIcon from '../../assets/img/sm_riot_icon.svg';
+import TwitterIcon from '../../assets/img/sm_twitter_icon.svg';
+import MediumIcon from '../../assets/img/sm_medium_icon.svg';
+import { ReactComponent as GearIcon } from '../../assets/img/gear-icon.svg';
+import { ReactComponent as StatIcon } from '../../assets/img/stat-icon.svg';
+
+import config from '../../config';
+const { twitter, gitHub, riot, medium, termsOfService, privacyPolicy } = config;
+
+const SideBar = ({ drawer, pathname }) => {
+    const classes = useSideBarStyles(drawer);
+    const onClick = (url) => {
+        if (url) {
+            window.open(url, '_blank');
+        }
+    };
+
+    const isActiveKeys = pathname === '/home/keys' ? '#7dc0de' : '#ffffff';
+    const isAciveStats = pathname === '/home/stats' ? '#7dc0de' : '#ffffff';
+
+    return (
+        <Drawer
+            className={classes.drawer}
+            variant="permanent"
+            classes={{
+                paper: classes.drawerPaper,
+            }}
+            anchor="left"
+        >
+            <div className={classes.toolbar}>
+                <CustomImg src={logo} size="3.875rem" name="logo" />
+            </div>
+            <Divider />
+            <MainList>
+                <Link to="/home/keys">
+                    <ListItemLarge button>
+                        <IconLarge>
+                            <IconWrapper>
+                                <CustomIcon
```

Comment:
It may make sense to create components for instances of CustomIcon with the same parameters.